packages/jazz-tools/src/tools/coValues/interfaces.ts

```diff
@@ -446,6 +448,92 @@ export function parseGroupCreateOptions(
     : { owner: options.owner ?? activeAccountContext.get() };
 }
 
+export function getIdFromHeader(
+  header: CoValueHeader,
+  loadAs?: Account | AnonymousJazzAgent | Group,
+) {
+  loadAs ||= activeAccountContext.get();
+
+  const node =
+    loadAs[TypeSym] === "Anonymous" ? loadAs.node : loadAs.$jazz.localNode;
+
+  return cojsonInternals.idforHeader(header, node.crypto);
+}
+
+export async function unstable_loadUnique<
+  V extends CoValue,
+  R extends RefsToResolve<V>,
+>(
+  cls: CoValueClass<V>,
+  options: {
+    header: CoValueHeader;
+    onCreateWhenMissing?: () => void;
+    onUpdateWhenFound?: (value: Resolved<V, R>) => void;
+    owner: Account | Group;
+    resolve?: RefsToResolveStrict<V, R>;
+  },
+): Promise<Resolved<V, R> | null> {
+  const loadAs = options.owner.$jazz.loadedAs;
+
+  const node =
+    loadAs[TypeSym] === "Anonymous" ? loadAs.node : loadAs.$jazz.localNode;
+
+  const id = cojsonInternals.idforHeader(options.header, node.crypto);
+
+  // We first try to load the unique value without using resolve and without
+  // retrying failures
+  // This way when we want to upsert we are sure that, if the load failed
+  // it failed because the unique value was missing
+  let result = await loadCoValueWithoutMe(cls, id, {
+    skipRetry: true,
+    loadAs,
+  });
+
+  if (options.onCreateWhenMissing) {
+    // if load returns unavailable, we check the state in localNode
+    // to ward against race conditions that would happen when
+    // running the same upsert unique concurrently
+    if (!result && node.getCoValue(id).hasVerifiedContent()) {
+      result = await loadCoValueWithoutMe(cls, id, {
+        loadAs,
+      });
+    }
+
+    if (!result) {
```

Comment:
What if the result is not available because the viewer is not authorized to read it? Should we create a new covalue?